re/lib/EsyBashLib.re

```diff
@@ -0,0 +1,111 @@
+exception InvalidEnvJSON(string);
+exception InvariantViolation(unit);
+
+let pathDelimStr = Sys.os_type == "Unix" ? "/" : "\\";
+let pathDelimChr = pathDelimStr.[0];
+
+let normalizePath = str =>
+  String.concat("/", String.split_on_char(pathDelimChr, str));
+
+let normalizeEndlines = str =>
+  String.concat("\n", Str.split(Str.regexp("\r\n"), str));
+
+let remapPathsInEnvironment = envVars =>
+  Array.map(
+    envVar =>
+      switch (String.split_on_char('=', envVar)) {
+      | [k, v, ...rest] =>
+        if (String.lowercase_ascii(k) == "path") {
+          "PATH=" ++ normalizePath(v);
+        } else {
+          k ++ "=" ++ v;
+        }
+      | _ => raise(InvariantViolation())
+      },
+    envVars,
+  );
+
+let collectKeyValuePairs = jsonKeyValuePairs =>
+  List.map(
+    pair => {
+      let (k, jsonValue) = pair;
+      switch (jsonValue) {
+      | `String(a) => k ++ "=" ++ a
+      | _ => raise(InvalidEnvJSON("Not a valid env json file"))
+      };
+    },
+    jsonKeyValuePairs,
+  );
+
+let rec traverse = json =>
+  switch (json) {
+  | `Assoc(keyValuePairs) => collectKeyValuePairs(keyValuePairs)
+  | _ => raise(InvalidEnvJSON("Not a valid env json file"))
+  };
+
+let extractEnvironmentVariables = environmentFile => {
+  let json = Yojson.Basic.from_file(environmentFile);
+  traverse(json);
+};
+
+let nonce = ref(0);
+let bashExec = (~environmentFile=?, command) => {
+  let shellPath =
+    Sys.os_type == "Unix" ? "/bin/bash" : "C:\\cygwin\\bin\\bash.exe"; /* "..\\.cygwin\\bin.bash.exe"; */
```

Comment:
This path looks hardcoded - we'll need to pick up a path relative to the executable

Reply:
We can do something like we do for `esy-bash` - go to a directory relative to the executable: https://github.com/esy/esy/blob/f31156f1c5385b5d3b19e2d812c1dd39228bc6eb/esy-lib/EsyBash.re#L14

In this case, we'd know exactly where the bash executable will be relative to our executable.